oauth-client-credentials/README.md

```diff
@@ -0,0 +1,115 @@
+
+## How it works
+
+With the client credentials grant type flow, the client app requests an access token directly by providing its client ID and client secret keys. These keys are generated when you create a Developer App in Apigee. Apigee validates the credentials and returns an access token to the client. The client can then make secure calls to the resource server.
+
+The API is called like this, where the client ID and secret are Base64-encoded and used in the Basic Auth header:
+
+```
+curl -H "Authorization: Basic <base64-encoded key:secret>" https://your-api-url.com/oauth/token?grant_type=client_credentials
+```
+
+## Implementation on Apigee 
+
+The client credentials sample uses one policy that executes on Apigee : An OAuthV2 policy to generate the access token. The policy is attached to the `/token` endpoint (a custom flow on Apigee). 
+
+## Prerequisites
+1. [Provision Apigee X](https://cloud.google.com/apigee/docs/api-platform/get-started/provisioning-intro)
+2. Configure [external access](https://cloud.google.com/apigee/docs/api-platform/get-started/configure-routing#external-access) for API traffic to your Apigee X instance
+3. Make sure the following tools are available in your terminal's $PATH (Cloud Shell has these preconfigured)
+    * [gcloud SDK](https://cloud.google.com/sdk/docs/install)
+    * unzip
+    * curl
+    * jq
+    * npm
+# (QuickStart) Setup using CloudShell
+
+Use the following GCP CloudShell tutorial, and follow the instructions.
+
+[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.png)](https://ssh.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/apigee-samples&cloudshell_git_branch=main&cloudshell_workspace=.&cloudshell_tutorial=oauth-client-credentials/docs/cloudshell-tutorial.md)
+
+## Setup instructions
+
+1. Clone the apigee-samples repo, and switch the oauth-client-credentials directory
+
+
+```bash
+git clone https://github.com/GoogleCloudPlatform/apigee-samples.git
+cd oauth-client-credentials
+```
+
+2. Edit the `env.sh` and configure the ENV vars
+
+* `PROJECT` the project where your Apigee organization is located
+* `APIGEE_HOST` the externally reachable hostname of the Apigee environment group that contains APIGEE_ENV
+* `APIGEE_ENV` the Apigee environment where the demo resources should be created
+
+Now source the `env.sh` file
+
+```bash
+source ./env.sh
+```
+
+3. Deploy Apigee API proxies, products and apps
+
+```bash
+./deploy-oauth-client-credentials.sh
+```
+
+## Testing the Client Credentials Proxy
+To run the tests, first retrieve Node.js dependencies with:
+```
+npm install
+```
+and then:
+```
+npm run test
```

Comment:
This might not work as the APIGEE_CLIENT_ID, SECRET and other variables are not set